service/service.go

```diff
@@ -192,6 +192,7 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 
 	tracerProvider, err := telFactory.CreateTracerProvider(ctx, telset, &cfg.Telemetry)
 	if err != nil {
+		logger.Error("failed to create tracer provider", zap.Error(err))
```

Comment:
One alternative way of doing this is to do something like this after line 190:

```go
defer func(){
     if err != nil {
         logger.Error("error found during service initialization", zap.Error(err))
     }
}()
```

(the message is a bit less specific but at least we don't have to add a new log on each place

Reply:
either works, one thing i would ask though is if the defer func is used, please add a comment otherwise i'll ask myself the same "why is this a defer func() here" every time i come across the code :D